### PR TITLE
Chapter 8 Basic login

### PR DESCRIPTION
**Pull request by PhanTrungDuc**

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  end
+
+  def create
+    user = User.find_by email: params[:session][:email].downcase
+    if user && user.authenticate(params[:session][:password])
+      log_in user
+      redirect_to user
+    else
+      flash.now[:danger] = "Invalid email/password combination"
+      render "new"
```
Comment (tiennh1995) [inline app/controllers/sessions_controller.rb]: render :new

File: app/helpers/sessions_helper.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  def current_user
+    @current_user ||= User.find_by id: session[:user_id]
+  end
+
+  def logged_in?
+    !current_user.nil?
+  end
+  
+  def log_out
+    session.delete(:user_id)
```
Comment (tiennh1995) [inline app/helpers/sessions_helper.rb]: bỏ ()

File: app/models/user.rb [hunk 3]
```diff
@@ -7,4 +7,10 @@ class User < ApplicationRecord
     uniqueness: {case_sensitive: false}
   has_secure_password
   validates :password, presence: true, length: {minimum: 6}
+
+  def User.digest string
+    cost = ActiveModel::SecurePassword.min_cost ? BCrypt::Engine::MIN_COST :
+                                                  BCrypt::Engine.cost
```
Comment (tiennh1995) [inline app/models/user.rb]: ident

File: app/models/user.rb [hunk 4]
```diff
@@ -7,4 +7,10 @@ class User < ApplicationRecord
...
+    BCrypt::Password.create(string, cost: cost)
```
Comment (tiennh1995) [inline app/models/user.rb]: Bỏ ()

File: app/views/layouts/_header.html.erb [hunk 5]
```diff
@@ -5,7 +5,24 @@
...
+          <li class="dropdown">
+            <a href="#" class="dropdown-toggle" data-toggle="dropdown">
+              Account <b class="caret"></b>
+            </a>
+            <ul class="dropdown-menu">
+              <li><%= link_to "Profile", current_user %></li>
+              <li><%= link_to "Settings", "#" %></li>
+              <li class="divider"></li>
+              <li>
+                <%= link_to "Log out", logout_path, method: "delete" %>
```
Comment (nguyenhuutien) [inline app/views/layouts/_header.html.erb]: method :delete

**Pull request merged**